src/Resolver.js

```diff
@@ -276,15 +276,37 @@ class Resolver {
     // libraries like d3.js specifies node.js specific files in the "main" which breaks the build
     // we use the "module" or "browser" field to get the full dependency tree if available.
     // If this is a linked module with a `source` field, use that as the entry point.
-    let main = [pkg.source, pkg.module, browser, pkg.main].find(
-      entry => typeof entry === 'string'
-    );
+    let main;
+    // this is a list possibile main files
+    let possibleMain = [pkg.main, browser, pkg.module, pkg.source];
+    // now we iter trough the list, from the end to the beginning.
+    // by doing this, we can easily pop elements from the list
+    for (let i = possibleMain.length - 1; i >= 0; i--) {
+      main = possibleMain[i];
+      if (typeof main !== 'string') {
+        // if the current main is not a string, trash it
+        possibleMain.pop();
+        continue;
+      }
+      try {
+        // else, check if the file main exists/is accessible. if it does, return the path of the file.
+        const resolvedPath = path.resolve(pkg.pkgdir, main);
+        await fs.access(resolvedPath);
```

Comment:
Isn't this the responsibility of the library anyone is using?

Shouldn't main always be accesible? Otherwise it's a bug in package.json from the pkg that's being processed

Reply:
>   // else, check if the file main exists/is accessible
here i'm referring to the main variable in general, not only the pkg.main file. The value of the main variable in that moment could be pkg.source. 
pkg.source is not always present in compiled packages, that's why parcel needs to check if it's accessible. If it's not, parcel should try to resolve the next possibleMain (e.g. pkg.module), until it finds an existing one.
Before this pr, parcel would try to load pkg.source even if it isn't accessible

Reply:
I think we should just have `getPackageMain` return an array of possible paths, and have `loadDirectory` loop through them and try each one. Since it is already reading files there, we would avoid checking twice.